src/utils.ts

```diff
@@ -1,50 +1,26 @@
-import { useCallback, useEffect, useState } from "react"
+import { useCallback, useState, useMemo } from "react"
 
 const EMPTY_ARRAY: any[] = []
 
-export function useUnmount(fn: () => void) {
-    useEffect(() => fn, EMPTY_ARRAY)
-}
-
 export function useForceUpdate() {
     const [, setTick] = useState(0)
 
     const update = useCallback(() => {
         setTick(tick => tick + 1)
-    }, [])
+    }, EMPTY_ARRAY)
 
     return update
 }
 
-export function isPlainObject(value: any): value is object {
-    if (!value || typeof value !== "object") {
-        return false
-    }
-    const proto = Object.getPrototypeOf(value)
-    return !proto || proto === Object.prototype
-}
+const deprecatedMessages: string[] = []
 
-export function getSymbol(name: string): symbol | string {
-    if (typeof Symbol === "function") {
-        return Symbol.for(name)
-    }
-    return `__$mobx-react ${name}__`
-}
-
-declare const window: any
-declare const self: any
-
-const mockGlobal = {}
-
-export function getGlobal() {
-    if (typeof window !== "undefined") {
-        return window
-    }
-    if (typeof global !== "undefined") {
-        return global
-    }
-    if (typeof self !== "undefined") {
-        return self
+export function useDeprecated(msg: string) {
+    if ("production" !== process.env.NODE_ENV) {
+        useMemo(() => {
```

Comment:
And that's how people optimize early :) Honestly, I don't know what you are trying to achieve here. Do you think that lookup in a nearly empty array is slower than magic behind memorization? :)

Reply:
Wow I did this way too late in the evening. The point wasn't optimization, I was jumping between two lines of thoughts: showing it once per component (using useMemo) or once for the app (using the messages list). I think the later is the nicer thing to do.

Reply:
I see. Warning per component would be a total nightmare, but that wouldn't happen anyway here since it uses a shared scoped variable.